### PR TITLE
remove us of malloc for nested_parallel

### PR DESCRIPTION
**Pull request by ronlieb**
. clang determines number of nested parallels per kernel
. level passed through the KernelDescriptor
. plugin rtl.cpp computes Teams*Threads*Levels just before launch
  and allocates memory for the callStack.
. serialized_parallel grabs its assigned omptarget_nvptx_TaskDescr item

Footnote: There are still uses of SafeMalloc in the runtimes.
  Tasks, data_env, and shared_args > MAX_SHARED_ARGS still use
  SafeMalloc which calls __malloc.

### CODE REVIEW

File: openmp/libomptarget/deviceRTLs/common/src/parallel.cu [hunk 1]
```diff
@@ -376,18 +378,30 @@ EXTERN void __kmpc_serialized_parallel(kmp_Ident *loc, uint32_t global_tid) {
...
   // it
-  omptarget_nvptx_TaskDescr *newTaskDescr =
-      (omptarget_nvptx_TaskDescr *)SafeMalloc(sizeof(omptarget_nvptx_TaskDescr),
-                                              "new seq parallel task");
+
+  // Each kernel has a precalculated call stack per thread.
+  // NumberTeams * NumberThreads * NumParallelLevels
+  // we calculate the max number of elements here
+  // Note that ParLev is the current parallel depth.
+  long CSIdx = blockDim.x * gridDim.x * ParLev;
```
Comment (JonChesterfield) [inline openmp/libomptarget/deviceRTLs/common/src/parallel.cu]: There are functions in support.h that wrap these cuda variables (and redirect to our intrinsics)

Comment (ronlieb) [reply openmp/libomptarget/deviceRTLs/common/src/parallel.cu]: oh, that explains it :-) , I will correct it.

**Comment by JonChesterfield** [comment]
The change looks good. It's also valid for nvptx though, so I think this should probably be implemented upstream and then ported back down.

File: openmp/libomptarget/deviceRTLs/common/src/parallel.cu (hunk 1 above)
Comment (ronlieb) [reply openmp/libomptarget/deviceRTLs/common/src/parallel.cu]: to be clear, are you suggesting I change blockDim.x to be GetNumberOfThreadsInBlock() ?
etc ...

Comment (JonChesterfield) [reply openmp/libomptarget/deviceRTLs/common/src/parallel.cu]: Indeed. Rather less succinct, but those were the existing names upstream.

**Comment by ronlieb** [comment]
> The change looks good. It's also valid for nvptx though, so I think this should probably be implemented upstream and then ported back down. Unfortunately that means patching the cuda plugin as well.

as this patch currently stands, i think it breaks nvpts nested-parallel.

so I think I may need to retain the use of SafeMalloc/SafeFree for the non-AMD build targets, in particular nvptx. To make this scheme work for targets other than AMD i will need to engineer a way to  pass in the parallel-level for that target.

**Pull request merged**